extension/llm/export/export_passes.py

```diff
@@ -95,3 +95,60 @@ def call(self, graph_module: torch.fx.GraphModule):
         graph_module.recompile()
 
         return PassResult(graph_module, graph_changed)
+
+
+class ReplaceSDPAWithCustomSDPAPass(ExportPass):
+    def call_operator(self, op, args, kwargs, meta):
+        from executorch.extension.llm.custom_ops import custom_ops  # noqa
+
+        if op != torch.ops.aten.scaled_dot_product_attention.default:
+            return super().call_operator(op, args, kwargs, meta)
+
+        q, k, v, *rest = args
+        mask = None
+        dropout = 0.0
+        is_causal = False
+        scale = None
+        if len(rest) > 0:
+            mask = rest[0]
+        if len(rest) > 1:
+            dropout = rest[1]
+        if len(rest) > 2:
+            is_causal = rest[2]
+        if "scale" in kwargs:
+            scale = kwargs["scale"]
+
+        qT = self._transpose(q, meta)
+        kT = self._transpose(k, meta)
+        vT = self._transpose(v, meta)
+
+        if mask is not None and mask.node.meta["val"].dtype == torch.bool:
```

Comment:
Put a todo here that custom sdpa once supports boolean mask, this wont be needed. tag me on the todo